common/achievement/get.ts

```diff
@@ -147,60 +147,80 @@ const assembleAchievementData = async (userAchievements: achievements_with_templ
+        ? userAchievements[currentAchievementIndex].template.condition.replace(
+              'recordValue',
+              (userAchievements[currentAchievementIndex].recordValue + 1).toString()
+          )
+        : userAchievements[currentAchievementIndex].template.condition;
 
     let maxValue: number = 0;
     let currentValue: number = 0;
-    if (currentAchievementTemplate.type === achievement_type_enum.STREAK || currentAchievementTemplate.type === achievement_type_enum.TIERED) {
-        const dataAggregationKeys = Object.keys(currentAchievementTemplate.conditionDataAggregations as Prisma.JsonObject);
+    if (
+        userAchievements[currentAchievementIndex].template.type === achievement_type_enum.STREAK ||
+        userAchievements[currentAchievementIndex].template.type === achievement_type_enum.TIERED
+    ) {
+        const dataAggregationKeys = Object.keys(userAchievements[currentAchievementIndex].template.conditionDataAggregations as Prisma.JsonObject);
         const evaluationResult = await evaluateAchievement(
             user.userID,
-            currentAchievementTemplate.condition,
-            currentAchievementTemplate.conditionDataAggregations as ConditionDataAggregations,
+            condition,
+            userAchievements[currentAchievementIndex].template.conditionDataAggregations as ConditionDataAggregations,
             userAchievements[currentAchievementIndex].recordValue || undefined,
             userAchievements[currentAchievementIndex].relation || undefined
         );
         if (evaluationResult) {
             currentValue = dataAggregationKeys.map((key) => evaluationResult.resultObject[key]).reduce((a, b) => a + b, 0);
             maxValue =
-                currentAchievementTemplate.type === achievement_type_enum.STREAK
-                    ? userAchievements[currentAchievementIndex].recordValue !== null && userAchievements[currentAchievementIndex].recordValue! > currentValue
-                        ? userAchievements[currentAchievementIndex].recordValue!
+                userAchievements[currentAchievementIndex].template.type === achievement_type_enum.STREAK
+                    ? userAchievements[currentAchievementIndex].recordValue !== null && userAchievements[currentAchievementIndex].recordValue > currentValue
+                        ? userAchievements[currentAchievementIndex].recordValue
                         : currentValue
                     : dataAggregationKeys
                           .map((key) => {
                               // TODO: check if we can remove valueToAchieve
-                              return Number((currentAchievementTemplate.conditionDataAggregations as any)[key].valueToAchieve as string);
+                              return Number(
+                                  (userAchievements[currentAchievementIndex].template.conditionDataAggregations as any)[key].valueToAchieve as string
```

Comment:
What if the field is not defined?

Reply:
Also here the || 0 operator was included.